shared_error.go

```diff
@@ -80,3 +86,40 @@ func (s *SharedError) Storef(format string, args ...interface{}) {
 
 	s.err = append(s.err, fmt.Errorf(format, args...))
 }
+
+// Err returns SharedError as an error when triggered.
+func (s *SharedError) Err() error {
+	if s.Triggered() {
+		return s
+	} else {
+		return nil
+	}
+}
+
+// Errors returns SharedError errors if any.
+func (s *SharedError) Errors() []error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	return s.err
+}
+
+// Has checks if any error of SharedError contains target error through errors.Is.
+func (s *SharedError) Has(target error) bool {
+	for _, err := range s.Errors() {
+		if errors.Is(err, target) {
+			return true
+		}
+	}
+	return false
+}
+
+// HasOnly checks if all errors of SharedError contains target error through errors.Is.
+func (s *SharedError) HasOnly(target error) bool {
+	for _, err := range s.Errors() {
+		if !errors.Is(err, target) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Good idea, but I'd call these `IsAny` and `IsAll`, referring to `errors.Is`.

Reply:
Great call, updated 👍